zerver/tests/test_auth_backends.py

```diff
@@ -3175,6 +3175,16 @@ def test_wrong_password(self) -> None:
                                   dict(password="wrong"))
         self.assert_json_error(result, "Your username or password is incorrect.", 400)
 
+    def test_invalid_subdomain(self) -> None:
+        username = 'hamlet'
+        user = self.example_user(username)
+        self.login_user(user)
+        with mock.patch("zerver.views.auth.get_realm_from_request", return_value=None):
+            result = self.client_post("/json/fetch_api_key",
+                                      dict(username='hamlet',
```

Comment:
```suggestion
                                      dict(username=username,
```

That's the benefit of having `        username = 'hamlet'` at the start of the test - so we don't have to put the `'hamlet`' string in multiple, but just consistently use the variable.